pandas/_libs/tslib.pyx

```diff
@@ -802,228 +802,33 @@ class Timestamp(_Timestamp):
         return self + other
 
 
-_nat_strings = set(['NaT', 'nat', 'NAT', 'nan', 'NaN', 'NAN'])
 
+# ----------------------------------------------------------------------
+# NaT Construction
+# Here we patch the docstrings of NaT methods to match those of Timestamp
+from tslibs.nattype import (NaT, NaTType,
+                            _make_error_func, _make_nat_func,
+                            _nat_strings)
+from tslibs.nattype cimport _nat_scalar_rules, _checknull_with_nat
 
-def _make_nat_func(func_name, cls):
-    def f(*args, **kwargs):
-        return NaT
-    f.__name__ = func_name
-    f.__doc__ = getattr(cls, func_name).__doc__
-    return f
+# We patch these NaTType methods with the Timestamp docstrings.
+NaTType.astimezone = _make_error_func('astimezone', Timestamp)
+NaTType.fromordinal = _make_error_func('fromordinal', Timestamp)
 
+# _nat_methods
+NaTType.to_pydatetime = _make_nat_func('to_pydatetime', Timestamp)
 
-def _make_nan_func(func_name, cls):
-    def f(*args, **kwargs):
-        return np.nan
-    f.__name__ = func_name
-    f.__doc__ = getattr(cls, func_name).__doc__
-    return f
+NaTType.now = _make_nat_func('now', Timestamp)
+NaTType.today = _make_nat_func('today', Timestamp)
+NaTType.round = _make_nat_func('round', Timestamp)
```

Comment:
This is very messy. instead define a function *in* ``tslib.nattype`` which does this (IOW calls the _make_nat_func) and pass in the Timestamp.

Reply:
or even better since there are not that many I would just copy-paste the doc-strings and avoid this whole thing.

Reply:
> or even better since there are not that many I would just copy-paste the doc-strings and avoid this whole thing

I like it!